server/modules/survey/service/publish/jobs/categoriesValidationJob.js

```diff
@@ -12,15 +12,16 @@ class CategoriesValidationJob extends Job {
   }
 
   async execute (tx) {
-    const categories = await CategoryManager.fetchCategoriesBySurveyId(this.surveyId, true, false, tx)
+    const categories = await CategoryManager.fetchCategoriesAndLevelsBySurveyId(this.surveyId, true, true, tx)
 
-    this.total = categories.length
+    const categoriesArr = Object.values(categories)
```

Comment:
why don't you iterate with Object.entries?

Reply:
you need only the values, not the keys (uuids) of categories

Reply:
yes, but you first get the values and then iterate. what i meant you can get the entries only. then iterate over them as well as get the length